src/anndata/_core/anndata.py

```diff
@@ -419,12 +390,12 @@ def _init_as_actual(
 
         # check data type of X
         if X is not None:
-            for s_type in StorageType:
-                if isinstance(X, s_type.value):
-                    break
-            else:
+            if not isinstance(X, StorageType.classes()):
                 raise ValueError(
-                    f"X needs to be of one of numpy.ndarray, numpy.ma.core.MaskedArray, scipy.sparse.spmatrix, h5py.Dataset, zarr.Array, anndata.experimental.[CSC,CSR]Dataset, dask.array.Array, cupy.ndarray, cupyx.scipy.sparse.spmatrix, not {type(X)}."
+                    "X needs to be of one of numpy.ndarray, numpy.ma.core.MaskedArray, "
+                    "scipy.sparse.spmatrix, h5py.Dataset, zarr.Array, "
+                    "anndata.experimental.[CSC,CSR]Dataset, dask.array.Array, "
+                    f"cupy.ndarray, cupyx.scipy.sparse.spmatrix, not {type(X)}."
```

Comment:
maybe this check should be moved into `coerce_arrays`?

also TODO: construct error message from enum